### DISCUSSION TITLE
Support for Variable Fonts and OpenType features

### DISCUSSION THREAD

**Discussion by gorjious** [cables-gl/cables_docs#268] [Feature Requests] [answered]
I am curious if there is/will be support for [Variable Fonts](https://v-fonts.com/). There would be such fascinating typographic potential if there is an Op that parses all the axes of a variable font to be manipulated with sliders or any other means. 

It would also be helpful to have access to the various other [OpenType Features](https://docs.microsoft.com/en-us/typography/opentype/spec/featurelist), such as stylistic sets, small caps, contextual alternates, ligatures, etc.

**Comment by steam0r** [comment, marked as answer]
hey,

these new ops are basically a wrapper around this library, so we are kinda bound to their capabilities:

https://github.com/opentypejs/opentype.js

here is a few questions on their issue tracker regarding variables fonts:

https://github.com/opentypejs/opentype.js/issues?q=is%3Aissue+is%3Aopen+variable

**Reply by gorjious** [reply]
Ah I see. Looks like they still have a lot of open issues...

What about using CSS Properties  Op – would it be possible to access the [font-variation-settings](https://developer.mozilla.org/en-US/docs/Web/CSS/font-variation-settings) and [font-feature-settings](https://developer.mozilla.org/en-US/docs/Web/CSS/font-feature-settings)?

**Reply by pandrr** [reply]
yes

**Reply by steam0r** [reply]
looks good: 

https://caniuse.com/?search=font-feature-settings 
https://caniuse.com/?search=font-variation-settings

maybe give it a try?

**Reply by gorjious** [reply]
Alrighty. Diving into my first venture using cables.gl! 
Thanks

**Reply by kdklv** [reply]
Tbh, I havent seen so many cases of variable fonts being used in webgl/canvas setting. Usually are seen rendered in dom, controlled trough css, etc.
But a while ago i played around with the textmesh op to do so / modified it so you could control certain variable axes like width, weight and slant. It is super messy and heavy, but it worked. Pandur also helped me a bunch, but since my coding skills are limited i got stuck and gave up. 

in case anyone is interested I can share the op and explain what i did (and werent able to do) :)

**Reply by gorjious** [reply]
@kdklv I'd love to take a look at the op and hear about your process!
